server/bin/cmv

```diff
@@ -0,0 +1,446 @@
+
+            if !subscription['derivedProduct'].nil? && !subscription['derivedProduct'].empty? then
+                if subscription['derivedProduct']['id'] && !subscription['derivedProduct']['id'].empty? then
+                    product_ids << subscription['derivedProduct']['id']
+                end
+            end
+
+            subscription['providedProducts'].each do |product|
+                product_ids << product['id']
+            end
+
+            subscription['derivedProvidedProducts'].each do |product|
+                product_ids << product['id']
+            end
+        end
+
+        products = []
+
+        product_ids.each do |pid|
+            product = candlepin.get_product(pid)
+            products << product
+        end
+
+        org_data['products'] = products
+
+        return org_data
+    end
+end
+
+class SnapshotCommand < CMVCommand
+    def initialize
+        super('Creates a snapshot of the products, content and pools for the specified org(s)')
+    end
+
+    def execute(candlepin, orgs=[], options={})
+        # This may become problematic on deployments with huge amounts of data. If it breaks, maybe
+        # look into writing the snapshot smaller files and zipping them all together
+        snapshot = {}
+
+        if !File.writable?(File.exists?(options[:file]) ? options[:file] : File.dirname(options[:file])) then
+            raise "Cannot write to snapshot file: #{options[:file]}"
+        end
+
+        orgs.each do |org|
+            puts "Compiling data for org \"#{org}\""
+            snapshot[org] = self.compileOrgData(candlepin, org)
+        end
+
+        # write snapshot to file
+        file = File.new(options[:file], 'w+')
+        file.write(JSON.generate(snapshot))
+        file.close()
+
+        puts "Snapshot successfully captured and saved to file: #{file.path}"
+    end
+end
+
+class VerifyCommand < CMVCommand
+    def initialize
+        super('Verifies the current data for the specified org(s) matches the last/given snapshot')
+
+        @id_cols = {
+            'orgs' => 'id',
+            'orgs.products' => 'id',
+            'orgs.products.productContent' => 'content.id',
+            'orgs.products.productContent.content' => 'id',
+            'orgs.products.attributes' => 'name',
+            'orgs.consumers' => 'uuid',
+            'orgs.pools' => 'id',
+            'orgs.pools.attributes' => 'name',
+            'orgs.pools.branding' => 'productId',
+            'orgs.pools.providedProducts' => 'productId',
+            'orgs.pools.productAttributes' => 'name',
+            'orgs.pools.derivedProvidedProducts' => 'productId',
+            'orgs.pools.derivedProductAttributes' => 'name',
+            'orgs.subscriptions' => 'id',
+        }
+
+        exclude = [
+            'orgs.pools.productAttributes.updated',
+            'orgs.products.attributes.updated',
+            'orgs.products.updated',
+            'orgs.products.productContent.content.updated'
+        ]
```

Comment:
Are we going to exclude the things we know will appear different? (href, owner links on products, etc) Ideally I'd like to see it run without args and return that everything is fine. Although I guess that does influence the generic nature of this a bit...